### PR TITLE
BKNDLSS-28895 Leaflet Map

### PR DESCRIPTION
**Pull request by EXPORTER1700**

### CODE REVIEW

File: components/bl-leafletMap-component/component.json [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+      ]
+    }
+  ],
+  "actions": [
+    {
+      "id": "mapCentering",
+      "label": "Set map center on",
+      "inputs": [
+        {
+          "id": "coords",
```
Comment (Valodya) [inline components/bl-leafletMap-component/component.json:265]: would not be better to have two separate inputs `latitude` and `longitude`?

File: components/bl-leafletMap-component/component.json [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+    },
+    {
+      "id": "getMapCircles ",
+      "label": "Get all circles of",
+      "hasReturn": true,
+      "color": "#f76c6c"
+    },
+    {
+      "id": "getMapPolygons",
+      "label": "Get all polygons",
```
Comment (Valodya) [inline components/bl-leafletMap-component/component.json:308]: `Get All polygons of`?

File: components/bl-leafletMap-component/src/geoposition-button.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+export function GeopositionButton({ map, eventHandlers }) {
+  const geoMarker = useRef(null);
+  const positionIcon = useIcon(IconOptions.position);
+
+  const handleClick = useCallback(() => {
+    getGeolocation(map, geoMarker, positionIcon, eventHandlers);
+  }, [map]);
+
+  return (
+    <button className="control button geo-button" onClick={ handleClick }><PositionIcon/></button>
```
Comment (Valodya) [inline components/bl-leafletMap-component/src/geoposition-button.js:17]:
```
<button className="control button geo-button" onClick={ handleClick }>
   <PositionIcon/>
</button>
```

File: components/bl-leafletMap-component/src/helpers/actions.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  const { markers, circles, polygons } = component;
+
+  component.mapCentering = (coords) => {
+    const currentCoords = toCoordinates(coords);
+    const currentZoom = map.getZoom();
+
+    map.setView(currentCoords, currentZoom);
+  };
+
+  component.zoomControl = (value) => {
```
Comment (Valodya) [inline components/bl-leafletMap-component/src/helpers/actions.js:13]: `setMapZoom`?

File: components/bl-leafletMap-component/src/helpers/coordinates.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    return coords;
+  }
+
+  if (typeof coords === 'string' && regEx.test(coords)) {
+    return coords.split(',');
+  }
+
+  console.error('Leaflet Map: not valid coordinates');
+
+  return [40.6893, -74.0444];
```
Comment (Valodya) [inline components/bl-leafletMap-component/src/helpers/coordinates.js:14]: the default value is already in the property, let's return here `[0,0]`

File: components/bl-leafletMap-component/src/index.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    toggleFullscreen(containerRef.current, fullscreen, mapRef.current);
+  }, [fullscreen]);
+
+  if (!display) {
+    return null;
+  }
+
+  return (
+    <div
+      id={ id }
```
Comment (Valodya) [inline components/bl-leafletMap-component/src/index.js:93]: https://github.com/Backendless/ui-builder-library/pull/11#discussion_r923365961

File: components/bl-leafletMap-component/component.json [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+        {
+          "id": "coords",
+          "label": "coordinates"
+        }
+      ],
+      "color": "#f76c6c"
+    },
+    {
+      "id": "zoomControl",
+      "label": "Set zoom on ",
```
[the remark this answers is not in the document]
Comment (EXPORTER1700) [reply components/bl-leafletMap-component/component.json:273]: Sets the zoom in Backendless Logic to the specified value.

**Pull request merged**